extensions/pkg/controller/worker/genericactuator/actuator_reconcile.go

```diff
@@ -358,8 +390,29 @@ func (a *genericActuator) waitUntilWantedMachineDeploymentsAvailable(ctx context
 			if health.CheckMachineDeployment(&deployment) == nil {
 				numHealthyDeployments++
 			}
-			numDesired += deployment.Spec.Replicas
-			numUpdated += deployment.Status.UpdatedReplicas
+
+			if gardenerutils.IsMachineDeploymentStrategyManualInPlace(deployment.Spec.Strategy) {
+				oldMachineSetsTotalReplicas := 0
+				oldMachineSets := extensionsworkerhelper.GetOldMachineSets(machineSets, *latestMachineSet)
+				for _, oldMachineSet := range oldMachineSets {
+					oldMachineSetsTotalReplicas += int(oldMachineSet.Status.Replicas)
```

Comment:
Can we document here why we need this `oldMachineSetsTotalReplicas` and cannot rely on the `.status.Replicas` and `.status.UpdatedReplicas` of the machine deployment?